lms/djangoapps/courseware/tabs.py

```diff
@@ -48,7 +48,7 @@ def get_course_tab_list(course, user):
     # Majority case is no entrance exam defined
     course_tab_list = []
     for tab in xmodule_tab_list:
-        if entrance_exam_mode:
+        if entrance_exam_mode and not user_can_skip_entrance_exam(user, course.id):
```

Comment:
I feel like this override should be part of the logic which determines the state of entrance_exam_mode